### PR TITLE
[SPARK-36332][SHUFFLE] Cleanup RemoteBlockPushResolver log messages

### PR DESCRIPTION
**Pull request by venkata91**
### What changes were proposed in this pull request?
Cleanup `RemoteBlockPushResolver` log messages by using `AppShufflePartitionInfo#toString()` to avoid duplications. Currently this is based off of https://github.com/apache/spark/pull/33034 will remove those changes once it is merged and remove the WIP at that time.

### Why are the changes needed?
Minor cleanup to make code more readable.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
No tests, just changing log messages

### CODE REVIEW

**Comment by venkata91** [comment]
cc @Ngone51

**Comment by venkata91** [comment]
Since https://github.com/apache/spark/pull/33034 is merged now, I have rebased the changes and marked this for review. cc @Ngone51 @mridulm

File: common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java [hunk 1]
```diff
@@ -1169,6 +1152,12 @@ void closeAllFilesAndDeleteIfNeeded(boolean delete) {
       }
     }
 
+    @Override
+    public String toString() {
+      return String.format("%s shuffleId %s shuffleMergeId %s reduceId %s",
```
Comment (Ngone51) [inline common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java:1157]: nit: `AppId %s shuffleId %s....`

File: common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java [hunk 2]
```diff
@@ -1032,9 +1022,7 @@ public long getDataFilePos() {
     }
 
     public void setDataFilePos(long dataFilePos) {
-      logger.trace("{} shuffleId {} shuffleMergeId {} reduceId {} current pos {}"
-        + " update pos {}", appId, shuffleId, shuffleMergeId, reduceId, this.dataFilePos,
-        dataFilePos);
+      logger.trace("{} current pos {} update pos {}", toString(), this.dataFilePos, dataFilePos);
```
Comment (mridulm) [inline common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java:1025]: `toString()` -> `this`

**Review by mridulm — APPROVED** [review]
+CC @Ngone51

**Review by Ngone51 — APPROVED** [review]

**Comment by Ngone51** [comment]
GA passed, merged to master/3.2, thanks!

**Pull request closed without merging**